packages/flutter/lib/src/widgets/reorderable_list.dart

```diff
@@ -83,6 +84,37 @@ typedef ReorderCallback = void Function(int oldIndex, int newIndex);
 /// The returned value will typically be the [child] wrapped in other widgets.
 typedef ReorderItemProxyDecorator = Widget Function(Widget child, int index, Animation<double> animation);
 
+/// Used to provide drag boundaries during drag-and-drop reordering.
+///
+/// {@tool snippet}
+/// ```dart
+/// DragBoundary(
+///  child: CustomScrollView(
+///    slivers: <Widget>[
+///      SliverReorderableList(
+///        itemBuilder: (BuildContext context, int index) {
+///          return ReorderableDragStartListener(
+///            key: ValueKey<int>(index),
+///            index: index,
+///            child: Text('$index'),
+///          );
+///        },
+///        dragBoundaryProvider: (BuildContext context) {
```

Comment:
What do you think if we make this the default behavior, so that it's opt-out instead of opt-in?

Reply:
I have considered doing this. If the boundary widget is named `ReorderableDragBoundary`, I think it is feasible. `DragBoundary` is not suitable to be provided by default to `SliverReorderableList`, as they do not seem to have any direct connection.